src/shims/unix/freebsd/foreign_items.rs

```diff
@@ -26,17 +26,29 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
                 this.write_null(dest)?;
             }
-            "pthread_set_name_np" => {
+            // Old vs new name
+            "pthread_set_name_np" | "pthread_setname_np" => {
```

Comment:
According to the manpage these are not the same, they have different return types. So the implementation will also have to be different, and both variants need to be tested.

Reply:
In fact I ll remove the new names for now and focus on the current ones, not even libc crate have them.